### PR TITLE
Hesabu V2 Api

### PR DESCRIPTION
**Pull request by pjaspers**
This does 2 things:

1. Do not break the existing API
2. Add a new API

## Do not break the existing API

I've added a constraint `lib/api_constraint.rb` that will pass requests coming to `api` to the `V2` version of the API if they have the `Accept: application/vnd.api+json;version=2` header.

This means that any existing API requests will never be passed along to V2, so it should all just work. This also means that when this goes live across all dhis-es, we can configure the apps to target a higher and lower version by modifying the `version=2` header. It's not something I foresee happening, but it's also something I'd like to have when we need it.

There is one small catch with this. CORS. The `OPTIONS` request that CORS sends doesn't listen to the `Accept` header, it will be send without that. This means that currently the Api v1, is handling those options requests. It does that here:

```
      match "*path",
            controller:  "application",
            action:      "options",
            constraints: { method: "OPTIONS" },
            via:         [:options]
```

I've added the same one in the v2 seciont, but that doesn't seem to be called. I still need to check if this is browser specific.

## Add a new API

This uses `JSON-API` to return resources. I've kept the naming of the `orbf-rules-engine` gem and of this project currently. In the routes I do expose the new naming of `hesabu-manager`, I think this helps keep it as clear as we can.

This also means that a `V2:PackageSerializer` can include things that we would normally not expect in there.

### CODE REVIEW

**Comment by pjaspers** [comment]
Deployed to staging and fixed the failing tests (the rubocops I kind of don't agree with).

File: app/controllers/api/v2/sets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class SetsController < BaseController
+      def index
+        packages = current_project_anchor.project.packages
```
Comment (mestachs) [inline app/controllers/api/v2/sets_controller.rb]: I would have expected the that we pass a project_id here (cfr versioning)

File: config/routes.rb [hunk 2]
```diff
@@ -20,20 +20,35 @@
...
+            constraints: { method: "OPTIONS" },
+            via:         [:options]
+    end
+    scope module: :v2, constraints: ApiConstraints.new(version: 2) do
+      resource :project, only: [:show]
+      resources :org_units, only: [:index]
+      resources :sets, only: [:index, :show]
+      resources :simulations, only: [:index, :show]
+      get :simulation, to: "simulations#query_based_show"
+      resources :topics, only: [:index, :show]
```
Comment (mestachs) [inline config/routes.rb]: where is this controller ?

File: app/controllers/api/v2/sets_controller.rb (hunk 1 above)
Comment (pjaspers) [reply app/controllers/api/v2/sets_controller.rb]: I think past me was blissfully ignoring versioned projects. That said, I think the default for the setscontroller should be to return the latest one, but I should provide support for other versions of the project.

Comment (pjaspers) [reply app/controllers/api/v2/sets_controller.rb]: I think past me was blissfully ignoring versioned projects. That said, I think the default for the setscontroller should be to return the latest one, but I should provide support for other versions of the project.

Comment (pjaspers) [reply app/controllers/api/v2/sets_controller.rb]: (And sets is the next thing I'm working on, so I'll handle this in a future PR)

File: config/routes.rb (hunk 2 above)
Comment (pjaspers) [reply config/routes.rb]: This I started, but since the PR was already too big, it's not in this one, so will remove this line from the routes.

**Pull request merged**